go/vt/vttablet/grpctmclient/client.go

```diff
@@ -72,13 +76,40 @@ type Client struct {
 	rpcClientMap map[string]chan *tmc
 }
 
+type dialer interface {
```

Comment:
note that for the `cachedConnDialer` we don't implement `poolDialer`, because every call to dial by definition is "pooled"', just for a slightly different definition of pooled